src/Tasks/RoslynCodeTaskFactory/RoslynCodeTaskFactory.cs

```diff
@@ -572,7 +585,24 @@ internal static bool TryResolveAssemblyReferences(TaskLoggingHelper log, RoslynC
             // Transform the list of resolved assemblies to TaskItems if they were all resolved
             items = hasInvalidReference ? null : resolvedAssemblyReferences.Select(i => (ITaskItem)new TaskItem(i)).ToArray();
 
+            handlerAddedToAppDomain = (_, eventArgs) => TryLoadAssembly(directoriesToAddToAppDomain, new AssemblyName(eventArgs.Name).Name);
+            AppDomain.CurrentDomain.AssemblyResolve += handlerAddedToAppDomain;
+
             return !hasInvalidReference;
+
+            static Assembly TryLoadAssembly(List<string> directories, string name)
+            {
+                foreach (string directory in directories)
+                {
+                    string path = Path.Combine(directory, name + ".dll");
```

Comment:
This is different and less complicated than what we have in `CoreCLRAssemblyLoader`. Do we need the culture-subdirectory lookup and the check-that-name-actually-matches stuff here, too?

Reply:
I don't think so. As I understand RoslynCodeTaskFactory, for this case, you have to pass it the exact path to your assembly, in which case it isn't in a subfolder, or you would've had to provide that. I only turned it into a directory and back into a file here to prevent a search for (unrelated) Foo to match Bar.dll just because I was too permissive.

Reply:
That's true for the assembly you directly reference, but what about _its_ dependencies and satellites?

Reply:
It'll silently work as long as its dependencies are next to it. If you have dependencies that you don't copy to your output directory, we don't have a great way of finding them anyway.

Reply:
And if you're thinking I should use an AssemblyDependencyResolver to read a deps.json file, that feels like overkill here 🙂

Reply:
I am not suggesting that, I am suggesting this:

https://github.com/dotnet/msbuild/blob/18fc0b3dd9dfd7cbb83272f13de4a89e8aa30531/src/Shared/CoreCLRAssemblyLoader.cs#L148-L175

Note specifically the culture-subdirectory stuff, which breaks the "next-to-it" rule but is predictable.